lib/ardb/has_slug.rb

```diff
@@ -53,26 +52,28 @@ module InstanceMethods
 
       private
 
-      def reset_slug
-        self.send("#{self.class.ardb_has_slug_config[:attribute]}=", nil)
+      def reset_slug(attribute = nil)
+        attribute ||= DEFAULT_ATTRIBUTE
+        self.send("#{attribute}=", nil)
       end
 
-      def ardb_has_slug_generate_slug
-        attr_name = self.class.ardb_has_slug_config[:attribute]
-        slug_source = if !self.send(attr_name) || self.send(attr_name).to_s.empty?
-          self.instance_eval(&self.class.ardb_has_slug_config[:source_proc])
-        else
-          self.send(attr_name)
+      def ardb_has_slug_generate_slugs
+        self.class.ardb_has_slug_configs.keys.each do |attr_name|
```

Comment:
@kellyredding - You might do: `self.class.ardb_has_slug_configs.each do |attr_name, config|`. Then you don't have to lookup the config.